src/main/java/org/takes/tk/TkProxy.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Dragan Bozanovic (bozanovicdr@gmail.com)
+ * @version $Id$
+ * @since 0.25
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class TkProxy implements Take {
+
+    /**
+     * Target host to which requests are forwarded.
+     */
+    private final transient String host;
+
+    /**
+     * Ctor.
+     *
+     * <p>The {@code target} parameter takes the destination URL
+     *  to which requests are proxied. Some valid examples:
+     *  <li>example.com
+     *  <li>www.example.com
+     *  <li>example.com:8080
+     *  <li>example.com/x/y
+     * @param target Target to which requests are forwarded
+     */
+    public TkProxy(final String target) {
+        this.host = target;
+    }
+
+    @Override
+    public Response act(final Request req) throws IOException {
+        final RqHref.Base base = new RqHref.Base(req);
+        final String home = new RqHref.Smart(base).home().bare();
+        final String dest = String.format(
+            "http://%s/%s",
+            this.host,
+            base.href().toString().substring(home.length())
+        );
+        return this.response(home, dest, this.request(req, dest).fetch());
+    }
+
+    /**
+     * Creates the request to be forwarded to the target host.
+     *
+     * @param req Original request
+     * @param dest Destination URL
+     * @return Request to be forwarded
+     * @throws IOException If some problem inside
+     */
+    private com.jcabi.http.Request request(final Request req,
+        final String dest) throws IOException {
+        final String method = new RqMethod.Base(req).method();
+        com.jcabi.http.Request proxied = new JdkRequest(dest)
+            .method(method);
+        final RqHeaders.Base headers = new RqHeaders.Base(req);
+        for (final String name : headers.names()) {
+            if ("content-length".equals(name.toLowerCase(Locale.ENGLISH))) {
+                continue;
+            }
+            if (isHost(name)) {
+                proxied = proxied.header(name, this.host);
+                continue;
+            }
+            for (final String value : headers.header(name)) {
+                proxied = proxied.header(name, value);
+            }
+        }
+        if (Arrays.asList("POST", "PUT").contains(method)) {
+            final ByteArrayOutputStream output = new ByteArrayOutputStream();
+            new RqPrint(new RqLengthAware(req)).printBody(output);
+            proxied = proxied.body().set(output.toByteArray()).back();
+        }
+        return proxied;
+    }
+
+    /**
+     * Creates the response received from the target host.
+     *
+     * @param home Home host
+     * @param dest Destination URL
+     * @param rsp Response received from the target host
+     * @return Response
+     */
+    private Response response(final String home, final String dest,
+            final com.jcabi.http.Response rsp) {
```

Comment:
@yegor256 Thanks for the remarks, please see my comments and new submitted changes.